pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitDaoImpl.java

```diff
@@ -666,4 +681,73 @@ public boolean hasKitReceived(Connection connection, String ddpParticipantId) {
         }
         return false;
     }
+
+    public ResultSet getKitsInDatabaseByInstanceId(DDPInstance ddpInstance) {
+        SimpleResult simpleResult = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SELECT_KIT_STATUS.concat(BY_INSTANCE_ID))) {
+                stmt.setString(1, ddpInstance.getDdpInstanceId());
+                ResultSet rs = stmt.executeQuery();
+                dbVals.resultValue = rs;
+            } catch (Exception ex) {
+                dbVals.resultException = new Exception(String.format("Error getting kits for %s", ddpInstance.getDdpInstanceId()));
+            }
+            return dbVals;
+        });
+        if (simpleResult.resultException != null) {
+            throw new DSMBadRequestException(simpleResult.resultException);
+        }
+        return (ResultSet) simpleResult.resultValue;
+
+    }
+
+    @Override
+    public ResultSet getKitsByJuniperKitId(String juniperKitId) {
+        SimpleResult simpleResult = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SELECT_KIT_STATUS.concat(BY_JUNIPER_KIT_ID))) {
+                stmt.setString(1, juniperKitId);
+                ResultSet rs = stmt.executeQuery();
+                dbVals.resultValue = rs;
+            } catch (Exception ex) {
+                dbVals.resultException = new Exception(String.format("Error getting kits with juniper kit id %s", juniperKitId));
+            }
+            return dbVals;
+        });
+        if (simpleResult.resultException != null) {
+            throw new DSMBadRequestException(simpleResult.resultException);
+        }
+        return (ResultSet) simpleResult.resultValue;
+    }
+
+    @Override
+    public ResultSet getKitsByParticipantId(String participantId) {
+        SimpleResult simpleResult = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SELECT_KIT_STATUS.concat(BY_PARTICIPANT_ID))) {
+                stmt.setString(1, participantId);
+                ResultSet rs = stmt.executeQuery();
+                dbVals.resultValue = rs;
+            } catch (Exception ex) {
+                dbVals.resultException = new Exception(String.format("Error getting kits with participant id %s", participantId));
+            }
+            return dbVals;
+        });
+        if (simpleResult.resultException != null) {
+            throw new DSMBadRequestException(simpleResult.resultException);
+        }
+        return (ResultSet) simpleResult.resultValue;
+    }
+
+    @Override
+    public ArrayList<NonPepperKitStatus> getKitsByKitIdArray(String[] kitIdsArray, NonPepperStatusKitService nonPepperStatusKitService) {
+        ArrayList<NonPepperKitStatus> list = new ArrayList<>();
```

Comment:
How large is this array typically? If large, perhaps a single query with post-processing would be more efficient?

Reply:
We decided it should not be larger than 20, if they need more, they will send the rest in another request